src/main.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	. "screepsgo/src/screeps"
+	"strconv"
+	"syscall/js"
+)
+
+//export Loop
+func Loop() {
+	var spawn = Game.Spawns()["Spawn1"]
+	var room = spawn.Room()
+
+	var myStructures = room.Find(FIND_MY_STRUCTURES)
+	var tower *StructureTower = nil
+	for _, v := range myStructures {
+		var structure = Structure{RoomObject: v} // TODO
+		if structure.StructureType() == STRUCTURE_TOWER {
+			tower = &StructureTower{OwnedStructure: OwnedStructure{Structure: structure}} // TODO
+			break
+		}
+	}
+	if tower != nil {
+		var structures = tower.Room().Find(FIND_STRUCTURES)
+		var damagedStructures = []RoomObject{} // TODO
+		for _, v := range structures {
+			var structure = Structure{RoomObject: v} // TODO
+			if structure.Hits() != nil && *structure.Hits() < *structure.HitsMax() {

```

Comment:
Do you need to dereference the structure pointer here? I believe that `stucture.Hits()` may work the same.

Reply:
I'm dereferencing `Hits()` and `HitsMax()` results (*int)